lib/scrabble_tilebag.rb

```diff
@@ -0,0 +1,49 @@
+require_relative './scrabble_scoring'
+# require_relative './scrabble_player'
+
+module Scrabble
+  class TileBag
+    attr_reader :all_tiles
+    attr_accessor :picked_tiles
+    def initialize
+      @all_tiles = {
```

Comment:
I like the idea of this data representation - it's very easily human readable.

However, storing it in a hash like this makes your `draw_tiles` method very complex.

A cleaner strategy might be to keep the default tiles outside the `initialize` method as a constant, and make a copy of it inside `initialize`. Something like

```ruby
class TileBag
  DEFAULT_TILES = { A: 9, B: 2, ... }
  def initialize
    @tiles = []
    DEFAULT_TILES.each do |letter, frequency|
      @tiles += [letter] * frequency
    end
  end
end
```

Then `@tiles` is a big array of all the tiles in the bag, which should make drawing from it much easier.